contracts/transmuter/libraries/LibOracle.sol

```diff
@@ -207,7 +244,80 @@ library LibOracle {
+    }
+
+    /// @notice Firewall in case the oracle value reported is too high compared to the target
+    /// @dev This mint firewall is useful in the case of assets for which the `targetPrice` is  theorically defined as
+    /// the maximum value ever observed for the oracle, but this maximum value has simply not been updated.
+    /// Typically, imagine a case where target is 100 and oracle is 101, in this setup, during a mint, because
+    /// target should be put at 101 but hasn't been modified, the system uses for a price the oracle value
+    function _firewallMint(uint256 targetPrice, uint256 oracleValue, uint256 deviation) private pure returns (uint256) {
+        if (targetPrice * (BASE_18 + deviation) < oracleValue * BASE_18) oracleValue = targetPrice;
+        return oracleValue;
+    }
+
+    /// @notice Firewall in case the oracle value reported is low compared to the target
+    /// @dev If the oracle value is slightly below its target, then no deviation is reported for the oracle and
+    /// the price of burning the stablecoin for other assets is not impacted
+    function _burnRatio(
+        uint256 targetPrice,
+        uint256 oracleValue,
+        uint256 deviation
+    ) private pure returns (uint256 ratio) {
+        ratio = BASE_18;
+        if (oracleValue * BASE_18 < targetPrice * (BASE_18 - deviation)) ratio = (oracleValue * BASE_18) / targetPrice;
+        return ratio;
+    }
+
+    /// @notice Firewall in case the oracle value reported is reasonably close to the target
+    /// --> disregard the oracle value and return the target price
+    function _userOracleProtection(
+        uint256 targetPrice,
+        uint256 oracleValue,
+        uint256 deviation
+    ) private pure returns (uint256) {
+        if (
+            targetPrice * (BASE_18 - deviation) < oracleValue * BASE_18 &&
+            oracleValue * BASE_18 < targetPrice * (BASE_18 + deviation)
+        ) oracleValue = targetPrice;
+        return oracleValue;
+    }
+
+    function updateOracle(address collateral) internal {
+        TransmuterStorage storage ts = s.transmuterStorage();
+        if (ts.collaterals[collateral].decimals == 0) revert NotCollateral();
+
+        (
+            OracleReadType oracleType,
+            OracleReadType targetType,
+            bytes memory oracleData,
+            bytes memory targetData,
+            bytes memory hyperparameters
+        ) = _parseOracleConfig(ts.collaterals[collateral].oracleConfig);
+
+        if (targetType != OracleReadType.MAX) revert OracleUpdateFailed();
+
+        uint256 oracleValue = read(oracleType, BASE_18, oracleData);
+        (uint256 maxValue, uint96 deviationThreshold, uint96 lastUpdateTimestamp, uint32 heartbeat) = abi.decode(
+            targetData,
+            (uint256, uint96, uint96, uint32)
+        );
+
+        if (
+            (oracleValue * BASE_18 >= maxValue * (BASE_18 + deviationThreshold)) ||
+            (block.timestamp - lastUpdateTimestamp > heartbeat)
```

Comment:
this is to avoid paying the gas if it's not needed? What's the rational for having a revert here instead of handling it at the keeper level?